services/streammanager/kafka/kafkamanager.go

```diff
@@ -450,14 +450,19 @@ func serializeAvroMessage(value []byte, codec goavro.Codec) ([]byte, error) {
 	return binary, nil
 }
 
-func prepareBatchOfMessages(topic string, batch []map[string]interface{}, timestamp time.Time, p producerManager) (
+func prepareBatchOfMessages(batch []map[string]interface{}, timestamp time.Time, p producerManager, defaultTopic string) (
 	[]client.Message, error,
 ) {
 	start := now()
 	defer func() { kafkaStats.prepareBatchTime.SendTiming(since(start)) }()
-
 	var messages []client.Message
+
 	for i, data := range batch {
+		topic, ok := data["topic"].(string)
+		if !ok || topic == "" {
+			topic = defaultTopic
```

Comment:
missing test coverage

Reply:
I have added this test case here already https://github.com/rudderlabs/rudder-server/blob/560fa4ef61f6d4cd7578c9309528c1e2bb2be047/services/streammanager/kafka/kafkamanager_test.go#L817

Reply:
This is addressed